docs/rbe-setup.md: use new Java toolchain flags (#5383)
buildbuddy-io/buildbuddy: docs/rbe-setup.md

### Java toolchain

> If your project depends on Java code, you'll need to set the following flags

Reviewer: ```suggestion
If your project depends on Java code, you'll need to set the following flags:
```

```bash
--java_language_version=11
--tool_java_language_version=11
--java_runtime_version=remotejdk_11
--tool_java_runtime_version==remotejdk_11
```

Available versions are listed in [Bazel's User Manual](https://bazel.build/docs/user-manual#java-language-version).

> If you need a custom Java toolchain, take a look at [Bazel and Java Docs](https://bazel.build/docs/bazel-and-java#config-java-toolchains)

Reviewer: ```suggestion
If you need a custom Java toolchain, see Bazel's docs on [Java toolchain configuration](https://bazel.build/docs/bazel-and-java#config-java-toolchains).
```

> #### Java toolchain for older Bazel

Reviewer: ```suggestion
#### Java toolchain for older Bazel versions
```

> If your project is using an older Bazel version (before Bazel 6.0.0), you will need the following 4 flags instead.

Reviewer: ```suggestion
If your project is using a Bazel version before 6.0.0, you will need the following 4 flags instead.
```
